### PR TITLE
gh-126594: Fix typeobject.c wrap_buffer() cast

### PR DESCRIPTION
**Pull request by vstinner**
Reject flags smaller than INT_MIN.

* Issue: gh-126594

### CODE REVIEW

**Comment by vstinner** [comment]
@JelleZijlstra: Do you want to review this change?

File: Objects/typeobject.c [hunk 1]
```diff
@@ -9314,13 +9314,13 @@ wrap_buffer(PyObject *self, PyObject *args, void *wrapped)
     if (flags == -1 && PyErr_Occurred()) {
         return NULL;
     }
-    if (flags > INT_MAX) {
+    if (flags > INT_MAX || flags < INT_MIN) {
         PyErr_SetString(PyExc_OverflowError,
                         "buffer flags too large");
```
Comment (JelleZijlstra) [inline Objects/typeobject.c]:
```suggestion
                        "buffer flags out of range");
```

**Review by JelleZijlstra — COMMENTED** [review]
Do you think it's worth adding a test? I understand this condition can perhaps only be triggered on some platforms.

**Comment by vstinner** [comment]
> Do you think it's worth adding a test? I understand this condition can perhaps only be triggered on some platforms.

Sure, I added a test.

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**